website/static/js/fileViewTreebeard.js

```diff
@@ -0,0 +1,110 @@
+var Fangorn = require('js/fangorn');
+var m = require('mithril');
+
+function FileViewTreebeard(data) {
+
+    // Set item.branch to show the branch of the rendered GitHub file instead of the default branch
+    var addonRootFolders = data['data'][0].children;
+    for (var i=0; i < addonRootFolders.length; i++) {
+        var item = addonRootFolders[i];
+        if (item.provider === 'github' && item.isAddonRoot && window.contextVars.file.extra.branch) {
+            item.branch = window.contextVars.file.extra.branch;
+        }
+    }
+
+    var fangornOpts = {
+    divID: 'grid',
+    filesData: data.data,
+    uploads: false,
+    showFilter: false,
+    title: undefined,
+    hideColumnTitles: true,
+    onload: function(tree) {
+        var tb = this;
+        Fangorn.DefaultOptions.onload.call(tb, tree);
+        $('.osf-panel-header.osf-panel-header-flex').show();
+    },
+    ondataload: function () {
+        var tb = this;
+        tb.options.folderIndex = 0;
+        if (window.contextVars.file.provider === 'figshare') {
+            tb.options.folderArray = [window.contextVars.file.name]
+        } else if (window.contextVars.file.path) {
+            window.contextVars.file.path = decodeURIComponent(window.contextVars.file.path);
+            tb.options.folderArray = window.contextVars.file.path.split("/");
+            if (tb.options.folderArray.length > 1) {
+                tb.options.folderArray.splice(0, 1);
+            }
+        } else {
+            tb.options.folderArray = [''];
+        }
+        m.render($('#files-search').get(0), tb.options.filterTemplate.call(tb));
```

Comment:
The m.render use is a limitation of mithril, search is a mithril template that can be placed anywhere on the page but it needs to be added in mithril style so that the template can be interpreted. This is normally handled in Treebeard but in this case Saman want's to put the search bar in a different position. 

This use is not inefficient or wrong, it's how mithril templates are rendered into html while maintaining their bindings.